packages/eslint-plugin-react-hooks/src/ExhaustiveDeps.js

```diff
@@ -223,6 +225,12 @@ export default {
         if (name === 'useRef' && id.type === 'Identifier') {
           // useRef() return value is stable.
           return true;
+        } else if (
+          (name === 'experimental_useEvent' || name === 'useEvent') &&
```

Comment:
Should this be behind a flag, only check experimental name, or only apply to improrts that are known from `react`? As it stands, I think it will flag all `useEvent` results even if it's from a third party library. Especially if we plan to later rename this Hook it seems like this could cause confusion and hide bugs. E.g. imagine you update and then suddenly your non-React `useEvent` deps are ignored by linter. Seems like overkill if we don't even end up using this name.

Reply:
I don't know if we have any infra in the eslint plugin to handle gated tests, so I dropped that idea. I'll update it to just check for the experimental name to avoid bloating the scope of this PR any further

Reply:
What would happen if you use `__EXPERIMENTAL__`? Since we bundle lint code too, wouldn't this work and make the logic gated to `@experimental` npm tag? Seems like that would be ideal

Reply:
I'll give that a shot!